src/go/guestagent/pkg/tracker/apitracker.go

```diff
@@ -152,9 +160,21 @@ func (a *APITracker) Remove(containerID string) error {
 				continue
 			}
 
-			log.Debugf("calling %s API for the following port binding: %+v", unexposeAPI, portBinding)
+			if a.enableListeners {
+				hostPort, err := strconv.Atoi(portBinding.HostPort)
+				if err != nil {
+					log.Errorf("error converting hostPort: %s", err)
+					continue
+				}
+				if err := a.RemoveListener(a.context, net.IP(portBinding.HostIP), hostPort); err != nil {
+					log.Errorf("removing listener for %s and %s failed: %s", portBinding.HostIP, portBinding.HostPort, err)
+					continue
+				}
+			}
+
+			log.Debugf("calling /services/forwarder/expose API for the following port binding: %+v", portBinding)
```

Comment:
Same as above:
```suggestion
			log.Debugf("unexposing the following port binding: %+v", portBinding)
```
(Note that this has a copy-and-paste error in the API path…)